fixit/common/config.py

```diff
@@ -95,35 +90,69 @@ def get_validated_settings(
         # Set path setting to absolute path.
         file_content[path_setting_name] = str(abspath)
 
+    file_content["inherit"] = file_content.get("inherit", False)
+    file_content["allow_list_rules"] = file_content.get("allow_list_rules", [])
+
     return file_content
 
 
 @lru_cache()
-def get_lint_config() -> LintConfig:
+def _get_config_schema() -> Dict[str, Any]:
+    # __package__ should never be none (config.py should not be run directly)
+    # But use .get() to make pyre happy
+    pkg = globals().get("__package__")
+    assert pkg, "No package was found, config types not validated."
+    config = pkg_resources.read_text(pkg, LINT_CONFIG_SCHEMA_NAME)
+    schema = json.loads(config)
+    return schema
+
+
+CACHE: Dict[Path, LintConfig] = {}
+
+
+def get_lint_config(path: Optional[Path] = None) -> LintConfig:
+    """
+    Get configuration for linting a given path. If a path is provided we walk up
+    the path and merge configurations along the way until `inherit` is set to `False`.
+    """
+    if path is None:
+        directory = Path.cwd()
+    elif not path.is_dir():
+        directory = path.parent
+    else:
+        directory = path
+
     config = {}
+    merger = Merger(_get_config_schema())
+
+    current = directory
+    while current.parent != current:
+        if current in CACHE:
+            return CACHE[current]
+
+        config_path = current / LINT_CONFIG_FILE_NAME
+
+        if config_path.is_file():
+            content = yaml.safe_load(config_path.read_text())
 
-    cwd = Path.cwd()
-    for directory in (cwd, *cwd.parents):
-        # Check for config file.
-        # pyre-fixme[58]: `/` is not supported for operand types `object` and `Path`.
-        possible_config = directory / LINT_CONFIG_FILE_NAME
-        if possible_config.is_file():
-            with open(possible_config, "r") as f:
-                file_content = yaml.safe_load(f.read())
+            if isinstance(content, dict):
```

Comment:
if there's a config file, but it's invalid (either because it's not yaml, or it's not a dict), we should assume `inherit` is the default value and `break`

This'd be easier to implement if we had a separate `get_default_settings` function which didn't take a `content` parameter